hs_collection_resource/models.py

```diff
@@ -0,0 +1,50 @@
+from django.contrib.auth.models import User
+from django.db import models, transaction
+
+from mezzanine.pages.page_processors import processor_for
+
+from hs_core.models import BaseResource, ResourceManager, resource_processor
+
+class CollectionResource(BaseResource):
+    objects = ResourceManager('CollectionResource')
+
+    class Meta:
+        proxy = True
+        verbose_name = 'Collection Resource'
+
+    @classmethod
+    def get_supported_upload_file_types(cls):
+        # no file types are supported
+        return ()
+
+    @classmethod
+    def can_have_multiple_files(cls):
+        # resource can't have any files
+        return False
+
+    @property
+    def can_be_public_or_discoverable(self):
+        return self.metadata.has_all_required_elements() and (self.resources.count() > 0)
```

Comment:
@zhiyuli @danames Should we enforce the collection has at least 2 resources before it can be made public? Having 1 resource in collection seems not sufficient in my opinion.